source/UF/SystemFNotionOfResizing.lagda

```diff
@@ -0,0 +1,348 @@
+    γ = 𝕣 (¬ (P holds)) λ _ → ⊥
+
+    P⁻ : 𝓤₀  ̇
+    P⁻ = resized (¬¬ (P holds)) γ
+
+    ζ : P holds → P⁻
+    ζ p = ⌜ ≃-sym (resizing-condition γ) ⌝ λ f → 𝟘-elim (f p)
+
+    ξ : P⁻ → P holds
+    ξ p⁻ = φ (eqtofun (resizing-condition γ) p⁻)
+
+    ψ : ¬¬ P⁻ → P⁻
+    ψ q = ζ (φ †)
+     where
+      † : ¬¬ (P holds)
+      † u = q (λ p⁻ → u (ξ p⁻))
+
+\end{code}
+
+The converse of this implication.
+
+\begin{code}
+
+  ν : (P : Ω¬¬ 𝓤₁) → P holds· → (r P) holds·
+  ν (P , φ) p = ⌜ ≃-sym (resizing-condition γ) ⌝ λ f → 𝟘-elim (f p)
+   where
+    γ : ¬¬ (P holds) is 𝓤₀ small
+    γ = 𝕣 (¬ (P holds)) λ _ → ⊥
+
+\end{code}
+
+We now combine these implications to get implications
+
+  - `r (s P) ⇔ P`, for every ¬¬-stable 𝓤₀-proposition `P`
+  - `s (r P) ⇔ P`, for every ¬¬-stable 𝓤₁-proposition `P`.
+
+\begin{code}
+
+  rs₁ : (P : Ω¬¬ 𝓤₀) → r (s P) holds· → P holds·
+  rs₁ P = ϑ P ∘ μ (s P)
+
+  rs₂ : (P : Ω¬¬ 𝓤₀) → P holds· → r (s P) holds·
+  rs₂ P = ν (s P) ∘ ι P
+
+  sr₁ : (P : Ω¬¬ 𝓤₁) → s (r P) holds· → P holds·
+  sr₁ P = μ P ∘ ϑ (r P)
+
+  sr₂ : (P : Ω¬¬ 𝓤₁) → P holds· → s (r P) holds·
+  sr₂ P = ι (r P) ∘ ν P
+
+\end{code}
+
+It follows easily from this then `Ω¬¬ 𝓤₀` is equivalent to `Ω¬¬ 𝓤₁`.
+
+\begin{code}
+
+  † : Ω¬¬ 𝓤₀ ≃ Ω¬¬ 𝓤₁
+  † = s , qinvs-are-equivs s (r , †₁ , †₂)
+   where
+    †₁ : (r ∘ resize-up-¬¬) ∼ id
+    †₁ (P , f) =
+     to-subtype-＝
+      (λ P → being-¬¬-stable-is-prop fe (holds-is-prop P))
+      (⇔-gives-＝ pe _ _ (holds-gives-equal-⊤ pe fe _ (rs₁ (P , f) , rs₂ (P , f))))
+
+    †₂ : resize-up-¬¬ ∘ r ∼ id
+    †₂ (P , f) =
+     to-subtype-＝
+      (λ Q → being-¬¬-stable-is-prop fe (holds-is-prop Q))
+      (⇔-gives-＝ pe _ P (holds-gives-equal-⊤ pe fe _ (sr₁ (P , f) , sr₂ (P , f))))
+
+\end{code}
+
+One might be tempted to consider Σ-resizing, but this immediately gives that 𝓤₀
+is 𝓤₀-small (thanks to Jon Sterling who pointed this out in a code review).
+
+\begin{code}
+
+Σ-Resizing : 𝓤₂  ̇
+Σ-Resizing = (A : 𝓤₁  ̇) → (B : A → 𝓤₀  ̇) → (Σ x ꞉ A , B x) is 𝓤₀ small
+
+Σ-resizing-gives-𝓤₀-is-𝓤₀-small : Σ-Resizing → (𝓤₀  ̇) is 𝓤₀ small
+Σ-resizing-gives-𝓤₀-is-𝓤₀-small ψ = resized ((𝓤₀  ̇) × 𝟙 {𝓤₀}) † , e
+ where
+  † : ((𝓤₀  ̇) × 𝟙 {𝓤₀}) is 𝓤₀ small
+  † = ψ (𝓤₀  ̇) (λ _ → 𝟙 {𝓤₀})
+
+  Ⅰ : resized ((𝓤₀  ̇) × 𝟙 {𝓤₀}) † ≃ (𝓤₀  ̇) × 𝟙 {𝓤₀}
+  Ⅰ = resizing-condition †
+
+  s : resized ((𝓤₀  ̇) × 𝟙 {𝓤₀}) † → 𝓤₀  ̇
+  s = pr₁ ∘ ⌜ Ⅰ ⌝
+
+  r : 𝓤₀  ̇ → resized ((𝓤₀  ̇) × 𝟙 {𝓤₀}) †
+  r A = ⌜ ≃-sym Ⅰ ⌝ (A , ⋆)
+
+  e : resized ((𝓤₀  ̇) × 𝟙 {𝓤₀}) † ≃ 𝓤₀  ̇
+  e = resized ((𝓤₀  ̇) × 𝟙 {𝓤₀}) †  ≃⟨ Ⅰ ⟩
+      (𝓤₀  ̇) × 𝟙 {𝓤₀}              ≃⟨ Ⅱ ⟩
+      𝓤₀  ̇                         ■
+       where
+        Ⅱ = 𝟙-rneutral {𝓤₁} {𝓤₀} {Y = (𝓤₀  ̇)}
+
+\end{code}
+
+The version of this notion of resizing with truncation, which we denote
+∃-resizing, must be consistent as it is implied by propositional resizing.
+
+\begin{code}
+
+∃-Resizing : 𝓤₂  ̇
+∃-Resizing = (A : 𝓤₁  ̇) → (B : A → 𝓤₀  ̇) → (Ǝ x ꞉ A , B x) holds is 𝓤₀ small
+
```

Comment:
Again consider `A` a proposition and `B x = 1` to conclude that this is equivalent to propositional resizing.

Reply:
Right, this one _is_ equivalent to propositional resizing.